plugin/entrySchema.go

```diff
@@ -172,6 +174,44 @@ func (s *EntrySchema) SetMetadataSchema(obj interface{}) *EntrySchema {
 	return s
 }
 
+// EntryType returns the entry's type
+func (s *EntrySchema) EntryType() string {
+	return s.entryType
+}
+
+// SetEntryType sets the entry's type. This should typically be the entry's class name.
+// Note that entry type is optional. It should only be set when you want analytics to
+// collect data on that entry.
+func (s *EntrySchema) SetEntryType(entryType string) *EntrySchema {
+	// We could try to get this information from the type ID; however, it is difficult
```

Comment:
I don't feel like this is useful enough for the added complexity. I don't agree that we need to leave out simple directories; it's easy enough to filter them out explicitly, or we can omit entries that don't have useful attributes/metadata and don't implement Read/Exec. I don't think capturing whether a plugin's types are used is something the plugin author should need to think about.

I'm also not sure the schema should be considered fragile. It's user-facing, so we shouldn't refactor it on a whim, and we won't be capturing long-running data.

Reply:
Open to alternative ways of getting the resource type, or omitting it altogether.

I do agree that knowing what resources are most popular doesn't really have any practical meaning aside from being something "cool to know." Plugin makes sense -- if AWS is popular, we should maintain AWS and maintain it well. That automatically includes its resources. So, I think I agree with you in that we should eliminate tracking the resource type altogether and just stick to method invocations (since the latter's simple enough to track).

Reply:
I'd be fine with that for now.

Reply:
Ok. I think it's still useful for the journal to record the method invocations, so will record that stuff via typeID instead of entryType.